### ISSUE TITLE
[Bug]: UnboundLocalError: local variable 'api_response' referenced before assignment

### ISSUE THREAD

**Issue by invideot01** [BlafKing/sd-civitai-browser-plus#155]
### Describe the bug.

When Laura, Checkpoint and etc. tries to scan, she gives an error, there is no Anti-Virus, and the firewall is configured to allow traffic for aria2.exe.

### Steps to reproduce the problem.

1. Go to Civitai Browser+
2. Go to Update Models
3. Check the "Scan for: Lora" 
4. Press button Updating model info&tags
5. Error

### Expected behavior

Everything should work correctly and start downloading information and tags from Civitai. (file.json)

### System info

* Extension version: 3.2.0
* OS: Win 10 Pro
* SD-WebUI version: 1.7.0
* Python: 3.10.11

### Console logs

```Shell
K:\StableDif>git pull
Already up to date.
venv "k:\StableDif\venv\Scripts\Python.exe"
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: v1.7.0
Commit hash: cf2772fab0af5573da775e7437e6acdca424f26e
Faceswaplab : Use GPU requirements
Checking faceswaplab requirements
0.007752000000436965
ReActor preheating... Device: CUDA
Launching Web UI with arguments: --theme=dark --opt-sdp-no-mem-attention --no-half-vae --xformers
Tag Autocomplete: Could not locate model-keyword extension, Lora trigger word completion will be limited to those added through the extra networks menu.
WARNING ⚠️ user config directory 'tmp\AppData\Roaming\Ultralytics' is not writeable, defaulting to '/tmp' or CWD.Alternatively you can define a YOLO_CONFIG_DIR environment variable for this path.
[-] ADetailer initialized. version: 23.9.2, num models: 17
CivitAI Browser+: Aria2 RPC started
2024-01-04 16:25:40,651 - ControlNet - INFO - ControlNet v1.1.422
ControlNet preprocessor location: K:\StableDif\extensions\sd-webui-controlnet\annotator\downloads
2024-01-04 16:25:40,822 - ControlNet - INFO - ControlNet v1.1.422
sd-webui-prompt-all-in-one background API service started successfully.
16:25:41 - ReActor - STATUS - Running v0.5.1-b4
Loading weights [87ad6c9624] from K:\StableDif\models\Stable-diffusion\SD\aimaginationEvolved_v10.safetensors
2024-01-04 16:25:42,743 - AnimateDiff - INFO - Hacking i2i-batch.
Creating model from config: K:\StableDif\configs\v1-inference.yaml
*Deforum ControlNet support: enabled*
Applying attention optimization: xformers... done.
No Image data blocks found.
No Image data blocks found.
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 30.9s (prepare environment: 6.9s, import torch: 4.0s, import gradio: 0.9s, setup paths: 0.9s, initialize shared: 0.2s, other imports: 0.7s, setup codeformer: 0.3s, list SD models: 0.7s, load scripts: 7.0s, create ui: 7.2s, gradio launch: 0.6s, app_started_callback: 1.3s).
Model loaded in 9.2s (load weights from disk: 1.0s, create model: 0.7s, apply weights to model: 2.2s, load textual inversion embeddings: 3.8s, calculate empty prompt: 1.4s).
Traceback (most recent call last):
  File "k:\StableDif\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "k:\StableDif\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "k:\StableDif\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "k:\StableDif\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "k:\StableDif\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "k:\StableDif\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "k:\StableDif\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "K:\StableDif\extensions\sd-civitai-browser-plus-main\scripts\civitai_file_manage.py", line 785, in file_scan
    api_response['items'] = all_items
UnboundLocalError: local variable 'api_response' referenced before assignment
```

### Additional information

On SD-WebUI version: 1.6.1 and Extension version: 2.1.0 all working good.

**Comment by CesarCruzUtec** [comment]
I get the same error while trying to update. Search and download works OK

```Error: Received status code 500 with URL: https://civitai.com/api/v1/models?limit=100&ids=37013&ids=110157&ids=139535&ids=25221&ids=122441&ids=147032&ids=151959&ids=155423&ids=106272&ids=156406&ids=80848&ids=79301&ids=22332&ids=33879&ids=162444&ids=184090&ids=244305&ids=36095&ids=35545&ids=105212&ids=31472&ids=113876&ids=88834&ids=11102&ids=124507&ids=109763&ids=102507&ids=77175&ids=124558&ids=4855&ids=50377&ids=4985&ids=186832&ids=158981&ids=8349&ids=28408&ids=21451&ids=98052&ids=97980&ids=104228&ids=74491&ids=21233&ids=52339&ids=130521&ids=73249&ids=15815&ids=155605&ids=149755&ids=176793&ids=187315&ids=130527&ids=156824&ids=11701&ids=32717&ids=144997&ids=12658&ids=8723&ids=15981&ids=147058&ids=43764&ids=60238&ids=23962&ids=129972&ids=15446&ids=10353&ids=5067&ids=81610&ids=31385&ids=63257&ids=247014&ids=99182&ids=27000&ids=56441&ids=28780&ids=122452&ids=92149&ids=21431&ids=9942&ids=27927&ids=10061&ids=20809&ids=110322&ids=111760&ids=118347&ids=132444&ids=21209&ids=25980&ids=106296&ids=141118&ids=61801&ids=20690&ids=116558&ids=124554&ids=48195&ids=124535&ids=140528&ids=10390&ids=107808&ids=13320&ids=25045&ids=71501&ids=27996&ids=115656&ids=4384&ids=86054&ids=28205&ids=137267&ids=16014&ids=14365&ids=17922&ids=65111&ids=21133&ids=67574&ids=122475&ids=58390&ids=30189&ids=97114&ids=108651&ids=155426&ids=113994&ids=61028&ids=87054&ids=58701&ids=52916&ids=132419&ids=23380&ids=83172&ids=85753&ids=117234&ids=11177&ids=122464&ids=4437&ids=118281&ids=6489&ids=124565&ids=129092&ids=17816&ids=28782&ids=27558&ids=77641&ids=16187&ids=83086&ids=11765&ids=57788&ids=73331&ids=106239&ids=246599&ids=12359&ids=8505&ids=80142&ids=88697&ids=81922&ids=31098&ids=87301&ids=104253&ids=4449&ids=87296&ids=20447&ids=15424&ids=87046&ids=127574&ids=141707&ids=124570&ids=115839&ids=49977&ids=24960&ids=129668&ids=135617&ids=16448&ids=27902&ids=33130&ids=47510&ids=160282&ids=145897&ids=6080&ids=28686&ids=129140&ids=109656&ids=111029&ids=104203&ids=128135&ids=87290&ids=20509&ids=43331&ids=38081&ids=32251&ids=118339&ids=12970&ids=18255&ids=161618&ids=8281&ids=90146&ids=124564&ids=25804&ids=8690&ids=92097&ids=114191&ids=12726&ids=27398&ids=21747&ids=124549&ids=32563&ids=30432&ids=83225&ids=79270&ids=107842&ids=23523&ids=84450&ids=146509&ids=4451&ids=132437&ids=43839&ids=117233&ids=132928&ids=27503&ids=37598&ids=75387&ids=124502&ids=156139&ids=124525&ids=244301&ids=20175&ids=144451&ids=24814&ids=60572&ids=75144&ids=27935&ids=88930&ids=88161&ids=45693&ids=157204&ids=13051&ids=138566&ids=28593&ids=96783&ids=122527&ids=9502&ids=92834&ids=30622&ids=90750&ids=144993&ids=45977&ids=65423&ids=27247&ids=116595&ids=149261&ids=173344&ids=87039&ids=124532&ids=40436&ids=86188&ids=51410&ids=133838&ids=153356&ids=163925&ids=55482&ids=8877&ids=6424&ids=114834&ids=118697&ids=17673&ids=144391&ids=147054&ids=111046&ids=23707&ids=134866&ids=145570&ids=5716&ids=96750&ids=153561&ids=70496&ids=151562&ids=54260&ids=33471&ids=102197&ids=50077&ids=13574&ids=178608&ids=31561&ids=115639&ids=124571&ids=17248&ids=13910&ids=196918&ids=134991&ids=27406&ids=107626&ids=135584&ids=27607&ids=28360&ids=12144&ids=158928&ids=167155&ids=92166&ids=50927&ids=161647&ids=6363&ids=23329&ids=20305&ids=25591&ids=178648&ids=94828&ids=196884&ids=42822&ids=85211&ids=24833&ids=84422&ids=16876&ids=76335&ids=19432&ids=130559&ids=141197&ids=81799&ids=163915&ids=54128&ids=28355&ids=80940&ids=19310&ids=151989&ids=106224&ids=4468&ids=36097&ids=9432&ids=52609&ids=40437&ids=156747&ids=90838&ids=117297&ids=246989&ids=28975&ids=246603&ids=36520&ids=20346&ids=155555&ids=132012
Traceback (most recent call last):
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "D:\IA\stable-diffusion-webui\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "D:\IA\stable-diffusion-webui\extensions\sd-civitai-browser-plus\scripts\civitai_file_manage.py", line 785, in file_scan
    api_response['items'] = all_items
UnboundLocalError: local variable 'api_response' referenced before assignment```

**Comment by tiphae** [comment]
Same since yesterday or so - search and download work okay, but all of the updates give an API timeout with the "api_response before assignment" errors shown above.

**Comment by BlafKing** [comment]
Thanks for reporting!

I've attempted to implement a fix in the latest version ([v3.2.2](https://github.com/BlafKing/sd-civitai-browser-plus/commit/e17e5ca6795d61d186706e6a7100372bcc860031))

Please let me know if the issue still exists! :)

**Comment by BlafKing** [comment]
I'll go ahead and set this bug report to resolved since there have been no reports for a while, please feel free to post another message here if you still experience issues! :)
